lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
+                   ON CONFLICT (guild_id)
+                   DO UPDATE SET role_id=EXCLUDED.role_id;"""
+        await interaction.client.pool.execute(query, role.guild.id, role.id)
+
+    @staticmethod
+    async def create_permission_overwrites(guild: discord.Guild, role: discord.Role):
+        success = 0
+        failure = 0
+        skipped = 0
+        for channel in guild.channels:
+            if channel.permissions_for(guild.me).manage_roles:
+                overwrite = channel.overwrites_for(role)
+                overwrite.read_messages = False
+                overwrite.send_messages = False
+                overwrite.add_reactions = False
+                overwrite.create_public_threads = False
+                overwrite.create_private_threads = False
+                overwrite.send_messages_in_threads = False
+                overwrite.use_application_commands = False
+                try:
+                    await channel.set_permissions(role, overwrite=overwrite,
+                                                  reason='Creating permission overwrites for the gatekeeper role')
+                except discord.HTTPException:
+                    failure += 1
+                else:
+                    success += 1
+            else:
+                skipped += 1
+        return success, failure, skipped
+
+    @discord.ui.button(label="Create a new role")
+    async def create_new_role(self, itx: discord.Interaction[LightningBot], button: discord.ui.Button):
+        try:
+            role = await itx.guild.create_role(name="Pending Verification",
+                                               reason=f"Requested role creation by {itx.user}")
+        except discord.Forbidden:
+            await itx.response.send_message("Unable to create a role because I am missing the Manage Roles permission!",
+                                            ephemeral=True)
+            return
+
+        await self.insert_role(itx, role)
+        view = ConfirmationView(message="", author_id=itx.user.id, delete_message_after=True)
+        await itx.response.edit_message(content="Created a new role. In order for this role to work "
+                                        "correctly, you must set permission overrides for every channel. "
+                                        "Would you like me to do this automatically?", view=view)
+        await view.wait()
+        if view.value is False:
+            # a message here
+            return
+
+        s, f, sk = await self.create_permission_overwrites(itx.guild, role)
+        if f >= 1:
+            content = f"Set {s+sk} permissions overwrites for this role. {f} channels failed to set permission "\
+                      "overrides!"
+        else:
+            content = f"Set {s+sk} permission overwrites for this role."
+
+        await itx.followup.send(content=content, ephemeral=True)
+        self.stop(interaction=itx)
+
+
+class GatekeeperChannelView(BasicMenuLikeView):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Add tests for `GatekeeperChannelView` to verify channel selection and permission setup.

Testing `GatekeeperChannelView` is essential to ensure that the verification channel is set correctly and that the permission overwrites for the channel are applied as intended.